packages/runtime-core/src/main/kotlin/elide/runtime/core/internals/graalvm/GraalVMEngine.kt

```diff
@@ -125,19 +177,43 @@ import org.graalvm.polyglot.HostAccess as PolyglotHostAccess
     ): GraalVMEngine {
       val languages = configuration.languages.map { it.languageId }.toTypedArray()
       val builder = Engine.newBuilder(*languages).apply {
+        useSystemProperties(false)
         allowExperimentalOptions(true)
 
+        // stub streams
+        `in`(StubbedInputStream)
+        out(StubbedOutputStream)
+        err(StubbedOutputStream)
+
+        // assign core log handler
+        logHandler(EngineLogHandler(engineLogger))
+
         // base options enabled for every engine
         enableOptions(
           "engine.BackgroundCompilation",
           "engine.UsePreInitializedContext",
           "engine.Compilation",
           "engine.MultiTier",
           "engine.Splitting",
-          "engine.Inlining",
           "engine.InlineAcrossTruffleBoundary",
+          "engine.OSR",
         )
 
+        // TODO(@darvld): swap for throughput in server mode
+        option("engine.Mode", "latency")
+
+        // TODO(@darvld): translate this to an API in the core package
+        listOfNotNull(
```

Comment:
This will later use a `HostRuntime` info API, similar to how `HostPlatform` works, so that plugins can also be version-aware. For now the current solution works really well, so no need to reimplement it here.